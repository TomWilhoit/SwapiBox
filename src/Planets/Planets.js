import React, { Component } from "react";
import "../Planets/Planets.scss";

class Planets extends Component {
  constructor() {
    super();
  }

  returnPlanets = () => {
    const planetItem = this.props.planetsArray.map((planet, index) => {
      return (
        <div key={index} className="card">
          <p>Name: {planet.name}</p>
          <p>Terrain: {planet.terrain}</p>
          <p>Population: {planet.population}</p>
          <p>Climate: {planet.climate}</p>
          <p>
            Residents:
            {planet.residents.map((resident, index) => {
              if (planet.residents.length - 1 === index) {
                return <span key={index}> {resident} </span>;
              } else {
                return <span> {resident},</span>;
              }
            })}
          </p>
        </div>
      );
    });
    return planetItem;
  };

  render() {
    if (this.props.planetsArray.length === 0) {
      return <div>LOADING</div>;
    } else {
      return (
        <div className="Planets">
          <h1 className="title">Planets</h1>
          <div className="button-container">
            <button className="btn" onClick={this.props.goToMain}>
              MAIN
            </button>
            <button className="btn" onClick={this.props.goToPeople}>
              PEOPLE
            </button>
            <button className="btn" onClick={this.props.goToVehicles}>
              VEHICLES
            </button>
          </div>
            <div className="card-container">{this.returnPlanets()}</div>
        </div>
      );
    }
  }
}

export default Planets;
