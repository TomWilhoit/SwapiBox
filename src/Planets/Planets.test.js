import React from "react";
import Planets from "../Planets/Planets";
import { shallow } from "enzyme";

describe("People", () => {
  let wrapper;
  let planetsArray = [
    {
        "name": "Alderaan",
        "rotation_period": "24",
        "orbital_period": "364",
        "diameter": "12500",
        "climate": "temperate",
        "gravity": "1 standard",
        "terrain": "grasslands, mountains",
        "surface_water": "40",
        "population": "2000000000",
        "residents": [
            "https://swapi.co/api/people/5/",
            "https://swapi.co/api/people/68/",
            "https://swapi.co/api/people/81/"
        ],
      }
    ]

  beforeEach(() => {
    wrapper = shallow(<Planets planetsArray={planetsArray} />);
  });

  it("should match snapshot when all data is passed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
