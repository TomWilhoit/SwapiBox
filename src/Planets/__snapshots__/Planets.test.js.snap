// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People should match snapshot when all data is passed correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Planets
    planetsArray={
      Array [
        Object {
          "climate": "temperate",
          "diameter": "12500",
          "gravity": "1 standard",
          "name": "Alderaan",
          "orbital_period": "364",
          "population": "2000000000",
          "residents": Array [
            "https://swapi.co/api/people/5/",
            "https://swapi.co/api/people/68/",
            "https://swapi.co/api/people/81/",
          ],
          "rotation_period": "24",
          "surface_water": "40",
          "terrain": "grasslands, mountains",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
        >
          Planets
        </h1>,
        <div
          className="button-container"
        >
          <button
            className="btn"
          >
            MAIN
          </button>
          <button
            className="btn"
          >
            PEOPLE
          </button>
          <button
            className="btn"
          >
            VEHICLES
          </button>
        </div>,
        <div
          className="card-container"
        >
          <div
            className="card"
          >
            <p>
              Name: 
              Alderaan
            </p>
            <p>
              Terrain: 
              grasslands, mountains
            </p>
            <p>
              Population: 
              2000000000
            </p>
            <p>
              Climate: 
              temperate
            </p>
            <p>
              Residents:
              <span>
                 
                https://swapi.co/api/people/5/
                ,
              </span>
              <span>
                 
                https://swapi.co/api/people/68/
                ,
              </span>
              <span>
                 
                https://swapi.co/api/people/81/
                 
              </span>
            </p>
          </div>
        </div>,
      ],
      "className": "Planets",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "className": "title",
        },
        "ref": null,
        "rendered": "Planets",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="btn"
            >
              MAIN
            </button>,
            <button
              className="btn"
            >
              PEOPLE
            </button>,
            <button
              className="btn"
            >
              VEHICLES
            </button>,
          ],
          "className": "button-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MAIN",
              "className": "btn",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "MAIN",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PEOPLE",
              "className": "btn",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "PEOPLE",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "VEHICLES",
              "className": "btn",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "VEHICLES",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="card"
            >
              <p>
                Name: 
                Alderaan
              </p>
              <p>
                Terrain: 
                grasslands, mountains
              </p>
              <p>
                Population: 
                2000000000
              </p>
              <p>
                Climate: 
                temperate
              </p>
              <p>
                Residents:
                <span>
                   
                  https://swapi.co/api/people/5/
                  ,
                </span>
                <span>
                   
                  https://swapi.co/api/people/68/
                  ,
                </span>
                <span>
                   
                  https://swapi.co/api/people/81/
                   
                </span>
              </p>
            </div>,
          ],
          "className": "card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Name: 
                  Alderaan
                </p>,
                <p>
                  Terrain: 
                  grasslands, mountains
                </p>,
                <p>
                  Population: 
                  2000000000
                </p>,
                <p>
                  Climate: 
                  temperate
                </p>,
                <p>
                  Residents:
                  <span>
                     
                    https://swapi.co/api/people/5/
                    ,
                  </span>
                  <span>
                     
                    https://swapi.co/api/people/68/
                    ,
                  </span>
                  <span>
                     
                    https://swapi.co/api/people/81/
                     
                  </span>
                </p>,
              ],
              "className": "card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Name: ",
                    "Alderaan",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Name: ",
                  "Alderaan",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Terrain: ",
                    "grasslands, mountains",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Terrain: ",
                  "grasslands, mountains",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Population: ",
                    "2000000000",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Population: ",
                  "2000000000",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Climate: ",
                    "temperate",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Climate: ",
                  "temperate",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Residents:",
                    Array [
                      <span>
                         
                        https://swapi.co/api/people/5/
                        ,
                      </span>,
                      <span>
                         
                        https://swapi.co/api/people/68/
                        ,
                      </span>,
                      <span>
                         
                        https://swapi.co/api/people/81/
                         
                      </span>,
                    ],
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Residents:",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "https://swapi.co/api/people/5/",
                        ",",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "https://swapi.co/api/people/5/",
                      ",",
                    ],
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "https://swapi.co/api/people/68/",
                        ",",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "https://swapi.co/api/people/68/",
                      ",",
                    ],
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": "2",
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "https://swapi.co/api/people/81/",
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "https://swapi.co/api/people/81/",
                      " ",
                    ],
                    "type": "span",
                  },
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          >
            Planets
          </h1>,
          <div
            className="button-container"
          >
            <button
              className="btn"
            >
              MAIN
            </button>
            <button
              className="btn"
            >
              PEOPLE
            </button>
            <button
              className="btn"
            >
              VEHICLES
            </button>
          </div>,
          <div
            className="card-container"
          >
            <div
              className="card"
            >
              <p>
                Name: 
                Alderaan
              </p>
              <p>
                Terrain: 
                grasslands, mountains
              </p>
              <p>
                Population: 
                2000000000
              </p>
              <p>
                Climate: 
                temperate
              </p>
              <p>
                Residents:
                <span>
                   
                  https://swapi.co/api/people/5/
                  ,
                </span>
                <span>
                   
                  https://swapi.co/api/people/68/
                  ,
                </span>
                <span>
                   
                  https://swapi.co/api/people/81/
                   
                </span>
              </p>
            </div>
          </div>,
        ],
        "className": "Planets",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": "title",
          },
          "ref": null,
          "rendered": "Planets",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="btn"
              >
                MAIN
              </button>,
              <button
                className="btn"
              >
                PEOPLE
              </button>,
              <button
                className="btn"
              >
                VEHICLES
              </button>,
            ],
            "className": "button-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MAIN",
                "className": "btn",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "MAIN",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PEOPLE",
                "className": "btn",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "PEOPLE",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "VEHICLES",
                "className": "btn",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "VEHICLES",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="card"
              >
                <p>
                  Name: 
                  Alderaan
                </p>
                <p>
                  Terrain: 
                  grasslands, mountains
                </p>
                <p>
                  Population: 
                  2000000000
                </p>
                <p>
                  Climate: 
                  temperate
                </p>
                <p>
                  Residents:
                  <span>
                     
                    https://swapi.co/api/people/5/
                    ,
                  </span>
                  <span>
                     
                    https://swapi.co/api/people/68/
                    ,
                  </span>
                  <span>
                     
                    https://swapi.co/api/people/81/
                     
                  </span>
                </p>
              </div>,
            ],
            "className": "card-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Name: 
                    Alderaan
                  </p>,
                  <p>
                    Terrain: 
                    grasslands, mountains
                  </p>,
                  <p>
                    Population: 
                    2000000000
                  </p>,
                  <p>
                    Climate: 
                    temperate
                  </p>,
                  <p>
                    Residents:
                    <span>
                       
                      https://swapi.co/api/people/5/
                      ,
                    </span>
                    <span>
                       
                      https://swapi.co/api/people/68/
                      ,
                    </span>
                    <span>
                       
                      https://swapi.co/api/people/81/
                       
                    </span>
                  </p>,
                ],
                "className": "card",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Name: ",
                      "Alderaan",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Name: ",
                    "Alderaan",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Terrain: ",
                      "grasslands, mountains",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Terrain: ",
                    "grasslands, mountains",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Population: ",
                      "2000000000",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Population: ",
                    "2000000000",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Climate: ",
                      "temperate",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Climate: ",
                    "temperate",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Residents:",
                      Array [
                        <span>
                           
                          https://swapi.co/api/people/5/
                          ,
                        </span>,
                        <span>
                           
                          https://swapi.co/api/people/68/
                          ,
                        </span>,
                        <span>
                           
                          https://swapi.co/api/people/81/
                           
                        </span>,
                      ],
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Residents:",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          " ",
                          "https://swapi.co/api/people/5/",
                          ",",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        " ",
                        "https://swapi.co/api/people/5/",
                        ",",
                      ],
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          " ",
                          "https://swapi.co/api/people/68/",
                          ",",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        " ",
                        "https://swapi.co/api/people/68/",
                        ",",
                      ],
                      "type": "span",
                    },
                    Object {
                      "instance": null,
                      "key": "2",
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          " ",
                          "https://swapi.co/api/people/81/",
                          " ",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        " ",
                        "https://swapi.co/api/people/81/",
                        " ",
                      ],
                      "type": "span",
                    },
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
